homeassistant/components/pilight/base_class.py

```diff
@@ -0,0 +1,186 @@
+"""Base class for pilight."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components import pilight
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.restore_state import RestoreEntity
+from homeassistant.const import (
+    CONF_ID,
+    CONF_NAME,
+    CONF_PROTOCOL,
+    CONF_STATE,
+    STATE_ON
+)
+
+from .const import (
+    CONF_OFF_CODE,
+    CONF_OFF_CODE_RECEIVE,
+    CONF_ON_CODE,
+    CONF_ON_CODE_RECEIVE,
+    CONF_SYSTEMCODE,
+    CONF_UNIT,
+    CONF_UNITCODE,
+    CONF_ECHO
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+COMMAND_SCHEMA = vol.Schema(
+    {
+        vol.Optional(CONF_PROTOCOL): cv.string,
+        vol.Optional("on"): cv.positive_int,
+        vol.Optional("off"): cv.positive_int,
+        vol.Optional(CONF_UNIT): cv.positive_int,
+        vol.Optional(CONF_UNITCODE): cv.positive_int,
+        vol.Optional(CONF_ID): vol.Any(cv.positive_int, cv.string),
```

Comment:
Why is the id a positive integer or a string?

Reply:
I found two switches in pilight with not only positive intergers as id:
https://manual.pilight.org/protocols/433.92/switch/clarus.html
https://manual.pilight.org/protocols/433.92/switch/x10.html

So I think a string should also be possible.

Reply:
Then I would suggest to changing it to only use string option.

Reply:
I don't prefer to change this, because that is a breaking change for existing configurations. It is acceptable to not change this?

Reply:
Yeah sure, leave it like this if changing it would create a breaking change.